packages/flutter_tools/lib/src/build_system/tools/asset_transformer.dart

```diff
@@ -22,13 +23,18 @@ final class AssetTransformer {
     required ProcessManager processManager,
     required FileSystem fileSystem,
     required String dartBinaryPath,
+    required BuildMode buildMode,
   })  : _processManager = processManager,
         _fileSystem = fileSystem,
-        _dartBinaryPath = dartBinaryPath;
+        _dartBinaryPath = dartBinaryPath,
+        _buildMode = buildMode;
+
+  static const String buildModeEnvVar = 'FLUTTER_BUILD_MODE';
```

Comment:
I have an internal instinct telling me that these should be namespaced, but `FLUTTER_ASSET_TRANFORMER__FLUTTER_BUILD_MODE` feels a bit verbose. I think we are fine here.